canarytokens/queries.py

```diff
@@ -133,44 +133,67 @@ def add_canary_nxdomain(domain: str) -> int:
     return DB.get_db().sadd(KEY_CANARY_NXDOMAINS, domain)
 
 
-def add_email_token_idx(email, canarytoken):
+def remove_set_member(key: str, member: str) -> None:
+    """
+    Removes `member` from the set at `key`,
+    and deletes the key if the set is empty afterwards.
+    """
+    current_set = DB.get_db().smembers(key)
+    DB.get_db().srem(key, member)
+    current_set.remove(member)
+    if current_set == set():
+        DB.get_db().delete(key)
```

Comment:
we're doing two things in this method where the method suggests it only does one thing "remove set member". perhaps we're getting a little fancy with the abstraction here.

there's also the risk of us nuking new items added to this list.

Lets say we:
- we remove a token (smembers list size is now 0)
- we read current_set into memory here and see it's zero
- user adds new token
- we delete the list 

we've not deleted the new token associated with the list.

I honestly wouldn't bother deleting the list from entirely here, it's one key / they might add more tokens in the future

Reply:
in fact i'd remove this method entirely and just use `DB.get_db().srem(key, member)` where we need it